presto-thrift/src/main/java/com/facebook/presto/thrift/clientproviders/PrestoClientProvider.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.thrift.clientproviders;
+
+import com.facebook.presto.spi.HostAddress;
+import com.facebook.presto.thrift.interfaces.client.ThriftPrestoClient;
+
+import java.util.List;
+
+public interface PrestoClientProvider
+{
+    ThriftPrestoClient connectToAnyHost();
```

Comment:
What kind of topologies do you mean? Currently, all thrift nodes are equal and independent. There is no coordinator for thrift nodes and they are not colocated with workers.